backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/query/EMFQueryService.java

```diff
@@ -51,8 +53,11 @@
 
     private final IEditingContextEPackageService editingContextEPackageService;
 
-    public EMFQueryService(IEditingContextEPackageService editingContextEPackageService) {
+    private final List<IQueryJavaServiceProvider> queryJavaServiceProviders;
+
+    public EMFQueryService(IEditingContextEPackageService editingContextEPackageService, List<IQueryJavaServiceProvider> queryJavaServiceProviders) {
```

Comment:
Minor, but an alternate constructor could have reduced the impacts:

```java
public EMFQueryService(IEditingContextEPackageService editingContextEPackageService) {
    this(editingContextEPackageService, List.of());
}
```

Reply:
Indeed :)

Reply:
Why would we need to keep the old constructor?

Reply:
To avoid impacting existing code which does not need to provide additional service classes.
Yes, it's only our own test code in this case, that's why I said it was a minor remark, not that it was a required change.